tests/nn_test.py

```diff
@@ -36,8 +39,111 @@
 
 config.parse_flags_with_absl()
 
-
+def _is_required_cudnn_version_satisfied():
+  if not jtu.test_device_matches(["cuda"]):
+    return False
+  if cuda_versions is None:
+    return False
+  if not jtu.is_cuda_compute_capability_at_least("8.0"):
+    return False
+  if cuda_versions.cudnn_get_version() < 8904:
+    return False
+
+  return True
+
+def _get_causal_mask(T, S):
+  assert T == S
```

Comment:
Why is this assert necessary?

Reply:
I think this is in case people changed the T, S values in the test. I think only when T == S, the causal mask is meaningful, right?

Reply:
No, I think you can meaningfully have a causal mask with T != S.

Reply:
Thanks. Removed the check.